tests/arraylib/array_lib.js

```diff
@@ -62,3 +62,16 @@ function of_test() {
   var upcastOkay: Array<string | number> = Array.of("hello", "world");
   var incompatibleTypeNotOkay: Array<string> = Array.of(1, 2);
 }
+
+function undefined_test() {
+  var emptyArray: Array<number> = []
+
+  var indexed: number = emptyArray[0] // error, undefined ~> number
```

Comment:
This line is actually passing, and I don't get why... 🤔 Any idea?

Reply:
Hmmm not sure either

Reply:
This is passing because indexer property access is also unsafe. As
@mrkev noted, this is by design, for the same reasons as with arrays.
See docs:

  - <https://flow.org/en/docs/types/objects/#toc-objects-as-maps>
  - <https://flow.org/en/docs/types/arrays/#array-access-is-unsafe->